### DISCUSSION TITLE
Novice, clumsy code? Post it yet?

### DISCUSSION THREAD

**Discussion by GlifTek** [community/community#23701] [New to GitHub]
<p>Hi,<br>
I’m working on posting my first repository.<br>
I’m wondering how, when a novice github user wants to post, has possibly novice, buggy code,<br>
How they preface this aspect?</p>
<p>I know in time my code will be perfected, yet I feel an urgency in getting it out there, regardless of perfection, as to the benefit it can bring to others.</p>
<p>So the question is:<br>
Can/should I post it whether it’s say…<br>
A mass of miscellaneous 3D objects I’d like people to make their own sense of, a Photoshop file full of layers with random descriptive measures applied to…</p>
<p>Ie: are there “rules” as to how things posted “have to be”?</p>
<p>Thanks!</p>

**Comment by kingthorin** [comment, marked as answer]
<p>There isn’t a rule.</p>
<p>I’d say as soon as you feel okay with people seeing it then go for it. Just be up-front about the fact that it’s early, and that you feel there are still adjustments/improvements to be made.</p>
<p>Also consider whether you want people’s feedback or not … repo issues on or off? Include an email address? etc. (Ex: careful what you ask for.)</p>

**Comment by GlifTek** [comment]
<p>Ok,<br>
I appreciate your response.<br>
I think your advice will put me in the right direction.<br>
Thanks!</p>
